### PR TITLE
Drop `V2`: Drop Modal component 

### PR DESCRIPTION
**Pull request by TuvalSimha**
### Before:

Screenshot 2024-06-23 at 19 54 47
Screenshot 2024-06-23 at 19 54 55

### After:

Screenshot 2024-06-23 at 19 54 04
Screenshot 2024-06-23 at 19 54 10

### Before:
Screenshot 2024-06-23 at 18 00 39

### After:
Screenshot 2024-06-23 at 18 12 40

### Before:
Screenshot 2024-06-23 at 18 02 41

### After:
Screenshot 2024-06-23 at 18 12 58

### Before:
Screenshot 2024-06-23 at 18 03 36

### After:
Screenshot 2024-06-23 at 18 13 17

### Before:
Screenshot 2024-06-23 at 18 04 22

### After:
Screenshot 2024-06-23 at 18 13 41

### Before:
Screenshot 2024-06-23 at 18 05 27

### After:
Screenshot 2024-06-23 at 18 16 25

### Before:
Screenshot 2024-06-23 at 18 06 06

### After:
Screenshot 2024-06-23 at 18 16 06

### Before:
Screenshot 2024-06-23 at 18 07 23

### After:
Screenshot 2024-06-23 at 18 16 16

### Before:
Screenshot 2024-06-23 at 18 08 04

### After:
Screenshot 2024-06-23 at 19 50 49

### Before:
Screenshot 2024-06-23 at 18 09 25

### After:
Screenshot 2024-06-23 at 19 45 44

### CODE REVIEW

**Review by dotansimha — APPROVED** [review]

File: packages/web/app/.storybook/preview.js [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
-if (!document.body.className.includes('dark')) {
-  document.body.className += ' ' + 'dark';
+//  Exposes environment variables to the frontend as a JavaScript object.
+window.__ENV ||= {
+  ENVIRONMENT: 'development',
+  APP_BASE_URL: 'http://localhost:3000',
+  GRAPHQL_PUBLIC_ENDPOINT: 'http://localhost:3001/graphql',
+  GRAPHQL_PUBLIC_SUBSCRIPTION_ENDPOINT: 'http://localhost:3001/graphql',
+  GRAPHQL_PUBLIC_ORIGIN: 'http://localhost:3001',
+};
```
Comment (dimaMachina) [inline packages/web/app/.storybook/preview.js:10]: Idk what a proper way to inject env variables (set `window.__ENV`) for the storybook, any suggestions?

File: packages/web/app/src/env/read.ts [hunk 2]
```diff
@@ -1,14 +1,3 @@
...
-}
-
 export function getAllEnv(): Record<string, string | undefined> {
-  if (isBrowser()) {
-    return (window as any)['__ENV'] ?? {};
-  }
-
-  throw new Error('getAllEnv should only be called in the browser');
+  return (window as any).__ENV ?? {};
 }
```
Comment (dimaMachina) [inline packages/web/app/src/env/read.ts:3]: https://github.com/kamilkisiela/graphql-hive/pull/5004/files#r1647826466

**Comment by kamilkisiela** [comment]
Screenshot 2024-07-10 at 10 55 07

Artifacts should not be part of the selectable options (the same applies to the other select component, contracts)
Screenshot 2024-07-10 at 10 26 36
8167190/e276963f-4882-4b63-99a2-e3c811d81e46)

Emails are missing here and the spacing is missing
Screenshot 2024-07-10 at 08 54 12

Creating a registry token does not work.
When I click on "Read only" option or pretty much any option, it has no effect.
Selecting 'No access' should not make "Generate Token" button active.
https://github.com/kamilkisiela/graphql-hive/assets/8167190/02b7c531-ed02-433d-9458-bbadff31b702

I will stop my review here.

Do you think splitting the PR into smaller PRs or doing it view by view, not component by component as I proposed before makes sense?

**Review by kamilkisiela — CHANGES_REQUESTED** [review]
I commented what needs to be improved

**Pull request closed without merging**